# Теория вероятностей и математическая статистика
## Урок 4. Непрерывная случайная величина

In [1]:
import numpy as np
from scipy import stats
import math

**Задача 1**  
Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800].
Найдите ее среднее значение и дисперсию.

In [2]:
A = (200, 800)

*Среднее значение*

In [3]:
# M(x)=(a+b)/2
Mx = (A[0]+A[1])/2
print(Mx)

500.0


*Диспресия*

In [4]:
# D(x)=((b-a)^2)/12
D = (A[1]-A[0])**2/12
print(D)

30000.0


-------------

**Ответ:**  
*Среднее значение* = 500  
*Дисперсия* = 30000

------------

**Задача 2**  
О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2.
Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5?
Если да, найдите ее.

In [5]:
a = 0.5 # левая граница
Dx = 0.2 # дисперсия

*Правая граница*

In [6]:
# Из формулы дисперсии -> b=sqrt(12D(x)) + a
b = math.sqrt(12*Dx) + a
round(b, 2)

2.05

*Среднее значение*

In [7]:
Mx = (a+b)/2
round(Mx, 2)

1.27

----------------

**Ответ:**  
*Правая граница* = 2.05  
*Среднее значение* = 1.27

------------

**Задача 3**  
Непрерывная случайная величина X распределена нормально и задана плотностью распределения

f(x) = (1 / (4 * sqrt(2*pi))) * (exp(-(x+2)**2) / 32).

Найдите:

а). M(X)

б). D(X)

в). std(X) (среднее квадратичное отклонение)

**Ответ:**  
Все значения даны в формуле:  
a) M(x) = -2   
б) D(x) = 16  
в) std(X) = 4

**Задача 4**  
Рост взрослого населения города X имеет нормальное распределение.

Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см.

Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:

In [8]:
a = 174 # средний рост
std = 8

*а). больше 182 см*

In [9]:
b182 = 182
z182 = (b182 - a) / std
z182

1.0

In [10]:
f182 = 0.8413 # значение функции распределения
p182 =  1 - f182
print(f'Вероятность того, что случайным образом выбранный взрослый человек имеет рост больше {b182}см  = {p182:.4f}')

Вероятность того, что случайным образом выбранный взрослый человек имеет рост больше 182см  = 0.1587


б). больше 190 см

In [11]:
b190 = 190
z190 = (b190 - a) / std
z190

2.0

In [12]:
f190 = 0.9772
p190 =  1 - f190
print(f'Вероятность того, что случайным образом выбранный взрослый человек имеет рост больше {b190}см  = {p190:.4f}')

Вероятность того, что случайным образом выбранный взрослый человек имеет рост больше 190см  = 0.0228


в). от 166 см до 190 см

In [13]:
b166 = 166
z166 = (b166 - a) / std
z166

-1.0

In [14]:
f166 =  0.1587
p166_190 =  f190 - f166
print(f'Вероятность того, что случайным образом выбранный взрослый человек имеет рост от {b166}см до {b190}см  = {p166_190:.4f}')

Вероятность того, что случайным образом выбранный взрослый человек имеет рост от 166см до 190см  = 0.8185


г). от 166 см до 182 см

In [15]:
f182 = 0.8413
p166_182 =  f182 - f166
print(f'Вероятность того, что случайным образом выбранный взрослый человек имеет рост от {b166}см до {b182}см  = {p166_182:.4f}')

Вероятность того, что случайным образом выбранный взрослый человек имеет рост от 166см до 182см  = 0.6826


д). от 158 см до 190 см

In [16]:
b158 = 158
z158 = (b158 - a) / std
z158

-2.0

In [17]:
f158 =  0.0228
p158_190 =  f190 - f158
print(f'Вероятность того, что случайным образом выбранный взрослый человек имеет рост от {b158}см до {b190}см  = {p158_190:.4f}')

Вероятность того, что случайным образом выбранный взрослый человек имеет рост от 158см до 190см  = 0.9544


е). не выше 150 см или не ниже 190 см

In [18]:
b150 = 150
z150 = (b150 - a) / std
z150

-3.0

In [19]:
f150 = 0.0013
p150_190 = f150 + 1 - f190
print(f'Вероятность того, что случайным образом выбранный взрослый человек имеет рост не выше {b150}см или не ниже {b190}см = {p150_190:.4f}')

Вероятность того, что случайным образом выбранный взрослый человек имеет рост не выше 150см или не ниже 190см = 0.0241


ё). не выше 150 см или не ниже 198 см

In [20]:
b198 = 198
z198 = (b198 - a) / std
z198

3.0

In [21]:
f198 = 0.9987
p150_198 = f150 + 1 - f198
print(f'Вероятность того, что случайным образом выбранный взрослый человек имеет рост не выше {b150}см или не ниже {b198}см = {p150_198:.4f}')

Вероятность того, что случайным образом выбранный взрослый человек имеет рост не выше 150см или не ниже 198см = 0.0026


ж). ниже 166 см.

In [22]:
print(f'Вероятность того, что случайным образом выбранный взрослый человек имеет рост ниже {b166}см  = {f166}')

Вероятность того, что случайным образом выбранный взрослый человек имеет рост ниже 166см  = 0.1587


**Задача 5**  
На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от
математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?


In [23]:
std = (190 - 178) / math.sqrt(25)
print(f'Рост человека, равный 190 см, отколняется на {std} сигмы')

Рост человека, равный 190 см, отколняется на 2.4 сигмы
